### PR TITLE
Fix targeting code

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Better xeno ability targeting code, should do what the players want 99% of the time while still allowing easier targeting

## Changelog
:cl:
code: Targeting code should allow players to use xeno abilities on the desired target more easily 
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
Super awesome change for xenos

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -307,16 +307,11 @@
...
-	if(.){\
-		return;}}
-
 /mob/living/carbon/xenomorph/proc/ability_target(atom/A)
-	TARGET_FLAGS_MACRO(XABB_MOB_TARGET, /mob/living)
-	if(selected_ability.target_flags & XABB_TURF_TARGET)
+	if(selected_ability.target_flags & XABB_MOB_TARGET && !isliving(A))
+		var/mob/living/target = locate(/mob/living) in get_turf(A)
+		return target ? target : A
+	if(selected_ability.target_flags & XABB_TURF_TARGET && !isturf(A))
```
Comment (BraveMole) [inline code/_onclick/click.dm:314]:
```suggestion
	if(selected_ability.target_flags & XABB_TURF_TARGET)
```

**Comment by SplinterGP** [comment]
can you keep the define ?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Use the define

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**